pkg/kubectl/cmd/delete.go

```diff
@@ -123,17 +124,6 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	}
 
 	ignoreNotFound := cmdutil.GetFlagBool(cmd, "ignore-not-found")
-	if deleteAll {
```

Comment:
Intuitively, if the user lists a bunch of objects by namespace, type, labels, fields, etc., then we want to ignore not-found errors by default, but if the objects are explicitly listed by name on the command line or via --filename, then we shouldn't ignore errors by default, but should provide an option to ignore errors.